packages/amplify-graphql-transformer-core/src/transformation/sync-utils.ts

```diff
@@ -47,6 +48,8 @@ function createSyncIAMRole(context: TransformerContext, scope: Construct, tableN
     roleName: context.resourceHelper.generateIAMRoleName(SyncResourceIDs.syncIAMRoleName),
     assumedBy: new iam.ServicePrincipal('appsync.amazonaws.com'),
   });
+  setResourceName(role.node.defaultChild!, SyncResourceIDs.syncIAMRoleName);
+  setResourceName(role, SyncResourceIDs.syncIAMRoleName);
```

Comment:
qq: why do we need to set the name on the `defaultChild` as well as the object itself?

Since this pattern is repeating, handling it centrally inside the `setResourceName` method with an optional flag `setOnChild` if required, can save us some lines of code.

Reply:
I think it is because it is setting the name for both L1 & L2 construct. But I agree to wrap this into one method to save the lines.